### PR TITLE
Refactor Arming Flags Priorities

### PR DESCRIPTION
**Pull request by giacomo892**
Do not display DISABLE NAVIGATION FIRST when for some reason there is no RC link on power up and channels are by default all at 1500 but display the real unable to arm reason aka NO RC LINK

### CODE REVIEW

File: src/main/navigation/navigation.c [hunk 1]
```diff
@@ -3226,7 +3226,7 @@ navArmingBlocker_e navigationIsBlockingArming(bool *usedBypass)
     }
 
     // Don't allow arming if any of NAV modes is active
-    if (!ARMING_FLAG(ARMED) && navBoxModesEnabled && !navLaunchComboModesEnabled) {
+    if (!ARMING_FLAG(ARMED) && !FLIGHT_MODE(FAILSAFE_MODE) && navBoxModesEnabled && !navLaunchComboModesEnabled) {
```
Comment (digitalentity) [inline src/main/navigation/navigation.c]: I would rather tweak the UI, not the logic, WDYT?

Comment (giacomo892) [reply src/main/navigation/navigation.c]: Or we might just refactor the RX input not to present 1500 when in FS. Toughts?

Comment (digitalentity) [reply src/main/navigation/navigation.c]: I would simply change the priorities of OSD messages. Internally we have both `NAV_IS_ALREADY_ACTIVE` and `NO_RC_LINK` triggered, the issue is only which one we want to show to the user first

Comment (giacomo892) [reply src/main/navigation/navigation.c]: I guess NO_RC_LINK should have the priority

Comment (giacomo892) [reply src/main/navigation/navigation.c]: I tried to swap around messages in osdArmingDisabledReasonMessage (osd.c) giving priority to 
RC LINK
ACC/MAG not calibrated
but still it displays navigation unusafe. 
Unsure, yet, what it is going on there

Comment (digitalentity) [reply src/main/navigation/navigation.c]: `osdArmingDisabledReasonMessage` merely displays the message. The actual reason (and only one reason) is returned by `isArmingDisabledReason` in runtime_config.c and that returns the first arming blocking bit set in `armingFlag_e`. 

You should probably swap bits for `ARMING_DISABLED_RC_LINK` (bit 18) and `ARMING_DISABLED_NAVIGATION_UNSAFE` (bit 11).

**Review by digitalentity — DISMISSED** [review]
LGTM

**Comment by stronnag** [comment]
3rd party consumers of MSP*STATUS* will not love you for this.

**Comment by digitalentity** [comment]
Oh indeed, damn. No LGTM then. We'll have to rewrite the `isArmingDisabledReason` to check for bits in the order we need.

**Comment by stronnag** [comment]
Or just bump the MW API version. That's fair warning.

**Comment by giacomo892** [comment]
It does indeed messes up also the `status` CLI output

**Comment by digitalentity** [comment]
Let's revert how it was then. I have an idea how to implement it in a non-breaking way

**Comment by giacomo892** [comment]
Superseeded by #5765

**Pull request closed without merging**